autorest/sender.go

```diff
@@ -207,6 +207,19 @@ func DoRetryForDuration(d time.Duration, backoff time.Duration) SendDecorator {
 	}
 }
 
+// WithLogging returns a SendDecorator that implements simple before and after logging of the
+// request.
+func WithLogging(logger *log.Logger) SendDecorator {
+	return func(s Sender) Sender {
+		return SenderFunc(func(r *http.Request) (*http.Response, error) {
+			logger.Printf("Sending %s %s\n", r.Method, r.URL)
```

Comment:
@ahmetalpbalkan http://play.golang.org/p/waiPl1eh7T.

fmt.Printf does *not* append a newline. fmt.Println, however, does.